libdrgn/dwarf_index.c

```diff
@@ -2357,11 +2413,28 @@ static struct drgn_error *index_cu_second_pass(struct drgn_dwarf_index *dindex,
 				file_name_hash = cu->file_name_hashes[decl_file - 1];
 			else
 				file_name_hash = 0;
-			if ((err = index_die(dindex, name,
-					     insn & DIE_FLAG_TAG_MASK,
+
+			struct drgn_dwarf_index_namespace *new_ns = NULL;
+			if ((err = index_die(ns, name,
+					     tag,
 					     file_name_hash, module,
-					     die_offset)))
+					     die_offset, &new_ns)))
 				return err;
+
+			if (new_ns) {
+				struct drgn_dwarf_index_pending_cu *pending;
+				#pragma omp critical(pending_cu_append)
+				{
+					if (pending = drgn_dwarf_index_pending_cu_vector_append_entry(
```

Comment:
Add extra parentheses around the assignment expression to make it clear that it's intentional and not a typo for `==`: `if ((pending = ...))`.